backend/src/main/java/wooteco/retrospective/dao/attendance/AttendanceDao.java

```diff
@@ -0,0 +1,73 @@
+package wooteco.retrospective.dao.attendance;
+
+import java.sql.PreparedStatement;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import wooteco.retrospective.dao.member.MemberDao;
+import wooteco.retrospective.domain.attendance.Attendance;
+
+@Repository
+public class AttendanceDao {
+
+    private final JdbcTemplate jdbcTemplate;
+    private MemberDao memberDao;
+    private TimeDao timeDao;
+
+    private final RowMapper<Attendance> rowMapper = (resultSet, rowNumber) ->
+        new Attendance(
+            resultSet.getLong("id"),
+            resultSet.getTimestamp("day"),
+            memberDao.findById(resultSet.getLong("member_id")).orElseThrow(RuntimeException::new),
+            timeDao.findById(resultSet.getLong("time_id")).orElseThrow(RuntimeException::new)
+        );
+
+    @Autowired
+    public AttendanceDao(JdbcTemplate jdbcTemplate, MemberDao memberDao,
+        TimeDao timeDao) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.memberDao = memberDao;
+        this.timeDao = timeDao;
+    }
+
+    public long insert(Attendance attendance) {
+        String query = "INSERT INTO ATTENDANCE(day, member_id, time_id) values (DEFAULT, ?, ?)";
+        KeyHolder keyHolder = new GeneratedKeyHolder();
+
+        jdbcTemplate.update(con -> {
+            PreparedStatement ps = con.prepareStatement(query, new String[] {"id"});
+            ps.setLong(1, attendance.getMember().getId());
+            ps.setLong(2, attendance.getTime().getId());
+
+            return ps;
+        }, keyHolder);
+
+        return Objects.requireNonNull(keyHolder.getKey()).longValue();
+    }
+
+    public Optional<Attendance> findById(long id) {
+        String query = "SELECT * FROM ATTENDANCE WHERE id = ?";
+
+        return Optional.ofNullable(jdbcTemplate.queryForObject(query, rowMapper, id));
+    }
+
+    public boolean isExistSameTime(long memberId, long timeId) {
+        String query = "SELECT * FROM ATTENDANCE WHERE member_id = ? AND time_id = ?";
+
+        return jdbcTemplate.query(query, rowMapper, memberId, timeId).size() > 0;
```

Comment:
```suggestion
        Attendance attendance = jdbcTemplate.query(query, rowMapper, memberId, timeId);
        return attendance.size() > 0;
```
이런 로직은 개인적으로 한번 끊어가는 편인데 어떠신가요? 가독성의 측면입니다! 사이즈가 무엇에 대한 사이즈를 뜻하는지 한번에 파악하기 어려워서요 :)

Reply:
그러네요 수정했습니다 ㅎㅎ